src/compositor/view.c

```diff
@@ -18,6 +18,13 @@ configure_view(struct wlc_view *view, uint32_t edges, const struct wlc_geometry
 {
    assert(view && g);
 
+   struct wlc_surface *surface = convert_from_wlc_resource(view->surface, "surface");
+   /* we need scale only for visible surface */
+   if(surface && surface->size.w * surface->size.h > 0) {
+      surface->scale.w = (float)(g->size.w) / surface->size.w;
+      surface->scale.h = (float)(g->size.h) / surface->size.h;
```

Comment:
Can the scale member be renamed to coordinate_scale or coordinate_transform perhaps, latter is usually used (it's usually matrix, but the type behind doesn't matter as long as we use the struct). Otherwise this can be easily confused that we are scaling surface, but we are actually just making input coordinates working with view which bounds aren't exact with the backing surface.